### PR TITLE
Add Language Server Tests Back In

### PR DESCRIPTION
**Pull request by ScottCarda-MS**
Reenables the Language Server Test project and tests. This PR follows many of the same changes as made in this PR: #1491, but fixes the issue of signing assemblies by avoiding changes to the DelaySign.props file. Here the language server is not created as a separate process, but connected to as an object instead to avoid having to build the language server separately.

### CODE REVIEW

**Comment by ScottCarda-MS** [comment]
The current QIR Validation failures seem to be unrelated to the change in the PR.

**Review by swernli — APPROVED** [review]
Minor feedback, but overall looks good. Thanks for digging into these!

File: src/QsCompiler/LanguageServer/Program.cs [hunk 1]
```diff
@@ -37,20 +37,27 @@ public class Options
                 HelpText = "Port to use for TCP/IP connections.")]
             public int Port { get; set; }
 
+            [Option(
+                "unnamed",
+                Required = false,
+                SetName = ConnectionViaPipe,
+                HelpText = "Connect via anonymous pipes.")]
+            public bool UseAnonymousPipes { get; set; }
```
[the remark this answers is not in the document]
Comment (ScottCarda-MS) [reply src/QsCompiler/LanguageServer/Program.cs]: I've changed this to internal. It doesn't have a purpose at this time beyond testing.

File: src/QsCompiler/LanguageServer/Program.cs [hunk 2]
```diff
@@ -37,20 +37,27 @@ public class Options
...
+
             [Option(
                 'w',
                 "writer",
                 Required = true,
                 SetName = ConnectionViaPipe,
-                HelpText = "Named pipe to write to.")]
+                HelpText = "Named or handle of the pipe to write to.")]
```
[the remark this answers is not in the document]
Comment (ScottCarda-MS) [reply src/QsCompiler/LanguageServer/Program.cs]: fixed

File: src/QsCompiler/LanguageServer/EditorState.cs [hunk 3]
```diff
@@ -545,7 +544,48 @@ internal Task CloseFileAsync(TextDocumentIdentifier textDocument, Action<string,
         internal Diagnostic[]? FileDiagnostics(TextDocumentIdentifier textDocument)
         {
             var allDiagnostics = this.projects.GetDiagnostics(textDocument?.Uri);
-            return allDiagnostics?.Count() == 1 ? allDiagnostics.Single().Diagnostics : null; // count is > 1 if the given uri corresponds to a project file
+            return allDiagnostics?.Length == 1 ? allDiagnostics.Single().Diagnostics : null; // count is > 1 if the given uri corresponds to a project file
+        }
+
+        /// <summary>
+        /// Waits for all currently running or queued tasks to finish before getting the project information.
```
[the remark this answers is not in the document]
Comment (ScottCarda-MS) [reply src/QsCompiler/LanguageServer/EditorState.cs:551]: The `GetProjectInformation` method waits for the tasks to finish before fetching the project information.

**Pull request merged**